app/src/main/java/com/example/csd228lab2/ui/viewmodels/DataStoresViewModel.kt

```diff
@@ -0,0 +1,124 @@
+package com.example.csd228lab2.ui.viewmodels
+
+import com.example.csd228lab2.AppSettings
+import android.content.Context
+import android.util.Log
+import androidx.datastore.core.CorruptionException
+import androidx.datastore.core.DataStore
+import androidx.datastore.core.Serializer
+import androidx.datastore.dataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.protobuf.InvalidProtocolBufferException
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import java.io.InputStream
+import java.io.OutputStream
+
+
+// Need to create two preferences using PreferenceDataStore and ProtoDataStore
+// Used to change the theme of the app to darkMode or lightMode
+class DataStoresViewModel(private val dataStore: DataStore<Preferences>,
+                          private val settingsDataStore: DataStore<AppSettings>) : ViewModel() {
```

Comment:
This would work great for testing purposes, I would recommend creating a viewModelFactory via:
```suggestion
class DataStoresViewModel(private val dataStore: DataStore<Preferences>,
                          private val settingsDataStore: DataStore<AppSettings>) : ViewModel() {
                          companion object Factory {
                              val Factory = viewModelFactory {
                                  initializer {
                                      val app = this[APPLICAITON_KEY] as MyCustomApplication // or whatever the name of your custom Application class is.
                                      DataStoresViewModel(dataStore = app.dataStore, settingsDataStore = app.settingsDataStore)
                                  }
                              }
                          }
```